pkg/antctl/raw/supportbundle/command.go

```diff
@@ -523,8 +580,54 @@ func controllerRemoteRunE(cmd *cobra.Command, args []string) error {
 	}
 	defer f.Close()
 	io.Copy(f, reader)
-	if err := requestAll(agentClients, controllerClient, bar); err != nil {
+
+	results := requestAll(agentClients, controllerClient, bar)
+	resultMap := downloadAll(agentClients, controllerClient, dir, bar, results)
+	return processResults(resultMap, dir)
+}
+
+func genErrorMsg(resultMap map[string]error) string {
+	msg := ""
+	for _, v := range resultMap {
+		msg += v.Error() + ";"
+	}
+	return msg
+}
+
+// processResults will output the failed nodes and their reasons if any. If no data was collected,
+// error is returned, otherwise will return nil.
+func processResults(resultMap map[string]error, dir string) error {
+	resultStr := ""
+	var failedNodes []string
+	allFailed := true
+	var err error
+
+	for k, v := range resultMap {
+		if k != "*" && v != nil {
+			resultStr += fmt.Sprintf("- %s: %s\n", k, v.Error())
+			failedNodes = append(failedNodes, k)
+		}
+		if v == nil {
+			allFailed = false
+		}
+	}
+
+	if resultMap["*"] != nil {
+		fmt.Println("Controller Info Failed Reason: " + resultMap["*"].Error())
+	}
+
+	if resultStr != "" {
+		fmt.Println("Failed nodes: ")
+		fmt.Print(resultStr)
+	}
+
+	if failedNodes != nil {
+		err = writeFailedNodes(dir, failedNodes)
```

Comment:
Also, it seems that failedNodes will never contain controller logs even when controller logs collection fails., in this way, we cannot record the controller failure via writeFailedNodes.

Reply:
the failedNodes file records k8s node names for now, since controller is not a node. Isn't it will cause misunderstaning for the users if we add controller failure to the failed nodes file?

Reply:
It makes sense. I was thinking if we use `antctl supportbundle` in another script and there is a chance that we can only get agents' logs but don't know why controller log is missing. Anyway, I am fine that we keep the current implementation.